src/java.base/share/classes/jdk/internal/foreign/StringSupport.java

```diff
@@ -53,6 +53,14 @@ public static String read(MemorySegment segment, long offset, Charset charset) {
         };
     }
 
+    public static String read(MemorySegment segment, long offset, int len, Charset charset) {
+        return switch (CharsetKind.of(charset)) {
+            case SINGLE_BYTE -> readByte(segment, offset, len, charset);
```

Comment:
How does this work, exactly? All methods called here seem to do:
```
byte[] bytes = new byte[len];
MemorySegment.copy(segment, JAVA_BYTE, offset, bytes, 0, len);
```

Why do we need different methods?